pkg/acsengine/defaults.go

```diff
@@ -471,7 +471,9 @@ func setMasterNetworkDefaults(a *api.Properties) {
 			if a.OrchestratorProfile.IsAzureCNI() {
 				// When VNET integration is enabled, all masters, agents and pods share the same large subnet.
 				a.MasterProfile.Subnet = a.OrchestratorProfile.KubernetesConfig.ClusterSubnet
-				a.MasterProfile.FirstConsecutiveStaticIP = getFirstConsecutiveStaticIPAddress(a.MasterProfile.Subnet)
+				if len(a.MasterProfile.FirstConsecutiveStaticIP) == 0 {
```

Comment:
What happens if the user provides a value for FirstConsecutiveStaticIP but not for Subnet?

Reply:
The current code will honor the FirstConsecutiveStaticIP if it is given by user. I think it is ok and if this value does not work, user can always remove it in their model and let it to be calculated automatically by the getFirstConsecutiveStaticIPAddress method.

Reply:
The current code overrides the FirstConsecutiveStaticIP if the subnet is invalid or nil. Adding that if statement without checking the subnet would mean a user could potentially have an apimodel with an a FirstConsecutiveStaticIP but no subnet which means that the IP and the subnet would not correspond since the subnet would still be the default one. Or provide both a FirstConsecutiveStaticIP and a ClusterSubnet and only ClusterSubnet would get overridden? (All of this is looking at it from the perspective of a first deployment, not upgrade). I don't think " if this value does not work, user can always remove it in their model" is good UX since we wouldn't be failing the apimodel validation.